src/network/network_content.cpp

```diff
@@ -70,9 +70,9 @@ bool ClientNetworkContentSocketHandler::Receive_SERVER_INFO(Packet *p)
 	ci->dependencies = MallocT<ContentID>(ci->dependency_count);
 	for (uint i = 0; i < ci->dependency_count; i++) ci->dependencies[i] = (ContentID)p->Recv_uint32();
 
-	ci->tag_count = p->Recv_uint8();
-	ci->tags = MallocT<char[32]>(ci->tag_count);
-	for (uint i = 0; i < ci->tag_count; i++) p->Recv_string(ci->tags[i], lengthof(*ci->tags));
+	uint tag_count = p->Recv_uint8();
+	ci->tags.reserve(tag_count);
```

Comment:
is `reserve` really useful? (honest question, to be clear)

Reply:
From cppreference: "Correctly using reserve() can prevent unnecessary reallocations, but inappropriate uses of reserve() (for instance, calling it before every push_back() call) may actually increase the number of reallocations...". Long story short, here we tell the vector: "heh, we know we need at least this many elements so you can prepare for that", and then the vector does not need to use its (potential) exponential growth algorithm where it doubles the size of the vector each time. So if there are say 9 tags, it would first make a vector of 2, then reallocate to 4, then reallocate to 8 and finally reallocate to 16. Since each reallocate might copy, that is potentially 14 unnecessary copies of data and leaving room for 7 tags that we know will never be filled.